### ISSUE TITLE
Neighbor does not automatically adjust for coordinate bounds

### ISSUE THREAD

**Issue by russellsteadman** [georust/geohash#48]
The `geohash::neighbor` function does not adjust when a coordinate reaches the latitude/longitude boundaries, so calculating neighbors throws errors for some hashes.

```rust
fn main() {
    let geohash = "2hb";

    let decoded = geohash::decode(geohash).unwrap();
    println!("{:?}", decoded.0.x_y());

    let neighbor = geohash::neighbor(geohash, geohash::Direction::W).unwrap();
    println!("{:?}", neighbor);
}
```

```
(-179.296875, -17.578125)
thread 'main' panicked at src\main.rs:7:70:
called `Result::unwrap()` on an `Err` value: InvalidCoordinateRange(Coord { x: -180.703125, y: -17.578125 })
```

Latitudes `y` range from `[-90.0, 90.0]` and longitudes `x` range from `[-180.0, 180.0]` where any value exceeding the upper bound should be added to the lower bound and vice versa.

```rust
let mut longitude: f64 = -181.0;
longitude = (longitude + 180.0).rem_euclid(360.0) - 180.0;
println!("{:?}", longitude);

let mut latitude: f64 = 91.0;
latitude = ((latitude + 90.0).rem_euclid(180.0)) - 90.0;
println!("{:?}", latitude);
```

```
179.0
-89.0
```

**Comment by sunng87** [comment]
Fixed in #49

**Comment by russellsteadman** [comment]
Would it be possible to push this bug fix as `0.13.1`? @sunng87

**Comment by sunng87** [comment]
Sure. `0.13.1` has been released to crates.io
